src/scenes/CharacterSelectionScene.ts

```diff
@@ -118,106 +120,104 @@ export class CharacterSelectionScene extends Scene<FriendlyFire> {
         animationTag: ControllerAnimationTags
     ): void {
-        const gap = 6;
+        const GAP = 6;
+        const controllerSprite = ControllerManager.getInstance().controllerSprite;
 
-        const textPositionX = Math.round(
-            x + this.controllerSpriteMapRecords[ControllerSpriteMap.KEYBOARD].width + gap
+        this.controllerSpriteMapRecords[controllerSprite].drawTag(
+            ctx,
+            animationTag,
+            position.clone()
         );
 
-        const textPositionY = y;
-        const controllerSprite = ControllerManager.getInstance().controllerSprite;
-        this.controllerSpriteMapRecords[controllerSprite].drawTag(ctx, animationTag, x, y)
-
         CharacterSelectionScene.font.drawTextWithOutline(
             ctx,
             text,
-            textPositionX, textPositionY,
-            "white",
-            "black"
+            position.clone().moveXBy(
+                this.controllerSpriteMapRecords[ControllerSpriteMap.KEYBOARD].width + GAP
+            ),
+            'white',
+            'black'
         );
     }
 
-    public draw(ctx: CanvasRenderingContext2D, width: number, height: number): void {
+    public draw(ctx: CanvasRenderingContext2D, size: Size): void {
         ctx.save();
 
         ctx.globalAlpha = 0.8;
         ctx.fillStyle = "black";
-        ctx.fillRect(0, 0, width, height);
+        ctx.fillRect(0, 0, size.width, size.height);
 
         ctx.globalAlpha = 1;
 
-        const x = (width / 2) - CharacterSelectionScene.panelImage.width / 2;
-        const y = (height / 2) - (CharacterSelectionScene.panelImage.height / 2) - 16;
+        const x = (size.width / 2) - CharacterSelectionScene.panelImage.width / 2;
+        const y = (size.height / 2) - (CharacterSelectionScene.panelImage.height / 2) - 16;
         ctx.translate(x, y);
 
         ctx.drawImage(CharacterSelectionScene.panelImage, 0, 0);
 
         CharacterSelectionScene.headlineFont.drawText(
-            ctx, "CHARACTER CUSTOMIZATION", 0, -14, "white"
+            ctx, "CHARACTER CUSTOMIZATION", new Point(0, -14), "white"
         );
 
         this.drawTooltip(
-            ctx, 0, CharacterSelectionScene.panelImage.height, "Select or Change",
+            ctx, new Point(0, CharacterSelectionScene.panelImage.height), "Select or Change",
             ControllerAnimationTags.CONFIRM
         );
 
         this.drawTooltip(
-            ctx, 0, CharacterSelectionScene.panelImage.height + 16, "Back",
+            ctx, new Point(0, CharacterSelectionScene.panelImage.height + 16), "Back",
             ControllerAnimationTags.BACK
         );
 
         const character = this.game.campaign.selectedCharacter;
-        const charSelectionTextY = characterMenuItemY + selectionItemsYDistance;
-        const charSelectionTextX = menuItemX + selectionItemsXDistance;
+        const charSelectionTextPosition = characterMenuItemPosition.clone().moveBy(selectionItemsOffset);
```

Comment:
Before the change: 0 objects per frame in this method. After the changes in this method: 10 objects per frame (Multiple clones and new Point instances)